lib/transport/transport-tls.c

```diff
@@ -190,7 +193,16 @@ log_transport_tls_write_method(LogTransport *s, const gpointer buf, gsize buflen
           errno = EAGAIN;
           break;
         case SSL_ERROR_SYSCALL:
-          /* errno is set accordingly */
+          // https://github.com/openssl/openssl/pull/11400
+          // There is a known bug in OpenSSL where it reports SSL_ERROR_SYSCALL without setting
+          // the proper errno value. The mentioned PR were reverted because lot of legacy code
+          // were broken by the fix. OpenSSL 3.0.0 will contain it.
+          if (errno == 0)
+            {
+              rc = -1;
+              errno = ECONNRESET;
+            }
+          rc = -1;
```

Comment:
Note: I would keep the original comment as well (`/* errno is set accordingly */`), as it is the intended behavior, which can't easily be expressed by code.